CorsixTH/Lua/endconditions.lua

```diff
@@ -0,0 +1,323 @@
+--!param table (table) the lose_criteria table for one group and one criteria.
+--!return If report is false, return the gap from hospital value to lose value, and the lose value
+-- if true, return the gap from the hospital value to boundary, and the boundary.
+function LoseCriteria.reputation(hospital, table, report)
+  local target = report and table.boundary or table.lose_value
+  local rep = hospital.reputation
+  local max_min = table.max_min_lose == 1 and 1 or -1
+  if (rep - target) * max_min >= 0 then
+    return math.abs(rep - target), target
+  end
+end
+
+function LoseCriteria.balance(hospital, table, report)
+  local target = report and table.boundary or table.lose_value
+  local bal = hospital.balance - hospital.loan
+  local max_min = table.max_min_lose == 1 and 1 or -1
+  if (bal - target) * max_min >= 0 then
+    return math.abs(bal - target), target
+  end
+end
+
+function LoseCriteria.percentage_killed(hospital, table, report)
+  local target = report and table.boundary or table.lose_value
+  local percentage_killed = hospital.percentage_killed
+  local max_min = table.max_min_lose == 1 and 1 or -1
+  if (percentage_killed - target) * max_min >= 0 then
+    return math.abs(percentage_killed - target), target
+  end
+end
+
+
+class "WinCriteria"
+--@type WinCriteria
+
+--!param hospital (Hospital) The hospital of the tests.
+--!param table (table) the win_criteria table for one group and one criteria
+--!return If report is false, return the gap from hospital value to win value, and the win value.
+-- if true, return the gap from the hospital value to boundary, and the boundary.
+function WinCriteria.reputation(hospital, table, report)
+  local target = report and table.boundary or table.win_value
+  local rep = hospital.reputation
+  local max_min = table.max_min_win == 1 and 1 or -1
+  if (rep - target) * max_min >= 0 then
+    return math.abs(rep - target), target
+  end
+end
+
+function WinCriteria.balance(hospital, table, report)
+  local target = report and table.boundary or table.win_value
+  local bal = hospital.balance - hospital.loan
+  local max_min = table.max_min_win == 1 and 1 or -1
+  if (bal - target) * max_min >= 0 then
+    return math.abs(bal - target), target
+  end
+end
+
+function WinCriteria.percentage_cured(hospital, table, report)
+  local target = report and table.boundary or table.win_value
+  local percentage_cured = hospital.percentage_cured
+  local max_min = table.max_min_win == 1 and 1 or -1
+  if (percentage_cured - target) * max_min >= 0 then
+    return math.abs(percentage_cured - target), target
+  end
+end
+
+function WinCriteria.num_cured(hospital, table, report)
+  local target = report and table.boundary or table.win_value
+  local num_cured = hospital.num_cured
+  local max_min = table.max_min_win == 1 and 1 or -1
+  if (num_cured - target) * max_min >= 0 then
+    return math.abs(num_cured - target), target
+  end
+end
+
+function WinCriteria.value(hospital, table, report)
```

Comment:
You may want to rename this one to `hospitalValue`, since `value` is quite generic.

Reply:
It uses the hospital attribute name. I'll add a comment to clarify that.